src/adapter/src/catalog.rs

```diff
@@ -7465,6 +7478,37 @@ impl SessionCatalog for ConnCatalog<'_> {
     fn all_object_privileges(&self, object_type: mz_sql_parser::ast::ObjectType) -> AclMode {
         rbac::all_object_privileges(object_type)
     }
+
+    fn get_object_type(&self, object_id: &ObjectId) -> mz_sql_parser::ast::ObjectType {
+        match object_id {
+            ObjectId::Cluster(_) => mz_sql_parser::ast::ObjectType::Cluster,
+            ObjectId::ClusterReplica(_) => mz_sql_parser::ast::ObjectType::ClusterReplica,
+            ObjectId::Database(_) => mz_sql_parser::ast::ObjectType::Database,
+            ObjectId::Schema(_) => mz_sql_parser::ast::ObjectType::Schema,
+            ObjectId::Role(_) => mz_sql_parser::ast::ObjectType::Role,
+            ObjectId::Item(item_id) => self.get_item(item_id).item_type().into(),
+        }
+    }
+
+    fn get_object_name(&self, object_id: &ObjectId) -> String {
+        match object_id {
+            ObjectId::Cluster(cluster_id) => self.get_cluster(*cluster_id).name().to_string(),
+            ObjectId::ClusterReplica((cluster_id, replica_id)) => self
```

Comment:
Same here. I guess ClusterReplica testing is still missing.

Reply:
ClusterReplicas don't have any associated privileges so we don't have any tests for them. This specific method is really a pain to write a rust test for and doesn't have much benefit, so I might just leave it out.